### PR TITLE
Restrict trigger for ARM to PRs containing labels only 

### PR DESCRIPTION
**Pull request by JimBobSquarePants**
### Prerequisites

- [x] I have written a descriptive pull-request title
- [x] I have verified that there are no overlapping [pull-requests](https://github.com/SixLabors/ImageSharp/pulls) open
- [x] I have verified that I am following the existing coding patterns and practice as demonstrated in the repository. These follow strict Stylecop rules :cop:.
- [x] I have provided test coverage for my change (where applicable)

### Description

Trying to keep the ARM build runner checks to a minimum to reduce costs.

Following a combination of the following instructions

https://stackoverflow.com/questions/62325286/run-github-actions-when-pull-requests-have-a-specific-label#comment122159108_62331521

https://github.com/orgs/community/discussions/26253#discussioncomment-3250989

### CODE REVIEW

**Comment by JimBobSquarePants** [comment]
First run with condition excludes ARM matrix entries. 

Now adding labels to ensure they are captured.

**Comment by JimBobSquarePants** [comment]
@brianpopow It works!!

File: .github/workflows/build-and-test.yml [hunk 1]
```diff
@@ -58,6 +61,10 @@ jobs:
             sdk: 6.0.x
             runtime: -x64
             codecov: false
+        exclude:
+          - isARM: false
```
Comment (brianpopow) [inline .github/workflows/build-and-test.yml:65]: @JimBobSquarePants: I am not sure that I understand the intention of the `exclude` here, does that mean only the arm runners should be executed and all others should not run?

Comment (JimBobSquarePants) [reply .github/workflows/build-and-test.yml:65]: The opposite. Exclude the ARM runners unless the label exists. 

I thought best to run the other runners at all times which helps us avoid any errors we potentially introduce with conditionals.

**Pull request merged**